debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotTableReader.java

```diff
@@ -0,0 +1,110 @@
+package io.debezium.connector.mysql;
+
+import io.debezium.function.BufferedBlockingConsumer;
+import io.debezium.jdbc.JdbcConnection;
+import io.debezium.relational.TableId;
+import io.debezium.util.Clock;
+import org.apache.kafka.connect.source.SourceRecord;
+
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Class snapshotting an individual table
+ * todo use a chainedReader to link all these individual readers together? maybe? is that what I want?
+ * the chained reader will make sure all of an individual reader's sourceRecords are polled before starting a new one.
+ * is that what I want?
```

Comment:
You tell me :) It's a bit of a general challenge for me that it's not quite clear to me where we're headed with this change.